src/Controller/ImportController.php

```diff
@@ -67,30 +82,347 @@ private function importDomainName(UploadedFile $csvFile, EntityManagerInterface
+                        'offer' => $sRow[1],
+                        'invoicedPrice' => $sRow[2],
+                        'cost' => $sRow[3],
+                        'renewalType' => $sRow[4],
+                        'startDate' => new \DateTime($sRow[5]),
+                        'renewalDate' => new \DateTime($sRow[6]),
+                    ];
+                    if ("" === $mail['renewalDate']) {
+                        $mail['renewalDate'] = date_add(new \DateTime($sRow[5]) , new DateInterval('P1Y'));
+                    }
+
+                    $futurMail = new Mail();
+                    $mailForm = $this->createForm(MailType::class, $futurMail);
+                    $mailForm->submit($mail);
+                    $mailForm->getData()->setStartDate($mail['startDate']);
+                    $mailForm->getData()->setRenewalDate($mail['renewalDate']);
+
+                    $mailForm->getData()->setClient($entityManager->getRepository(Client::class)->findOneBy(['name' => $sRow[7]]));
+                    $mailForm->getData()->setDomainName($entityManager->getRepository(DomainName::class)->findOneBy(['url' => $sRow[8]]));
+
+                    $entityManager->persist($mailForm->getData());
+
+                }
+            }
+            $entityManager->flush();
+        }
+    }
+
+    /**
+     * @param UploadedFile $csvFile
+     * @param EntityManagerInterface $entityManager
+     * @return void
+     * @throws \Exception
+     */
+    private function importSocialNetwork(UploadedFile $csvFile, EntityManagerInterface $entityManager) {
+
+        if ($csvFile && false !== ($fps = fopen($csvFile->getPath() . '/' . $csvFile->getFilename(), "r"))) {
+            $indexCSV = 0;
+            while (false !== ($sRow = fgetcsv($fps, 1000, ';'))) {
+                ++$indexCSV;
+                if (1 <= $indexCSV) {
+                    $socialNetwork = [
+                        'provider' => $sRow[0],
+                        'offer' => $sRow[1],
+                        'invoicedPrice' => $sRow[2],
+                        'cost' => $sRow[3],
+                        'renewalType' => $sRow[4],
+                        'startDate' => new \DateTime($sRow[5]),
+                        'renewalDate' => new \DateTime($sRow[6]),
+                        'postByWeek' => $sRow[7],
+                        'whichSocialNetwork' => explode(",", $sRow[8]),
+                    ];
+                    if ("" === $socialNetwork['renewalDate']) {
+                        $socialNetwork['renewalDate'] = date_add(new \DateTime($sRow[5]) , new DateInterval('P1Y'));
+                    }
+
+                    $futurSocialNetwork = new SocialNetwork();
+                    $socialNetworkForm = $this->createForm(SocialNetworkType::class, $futurSocialNetwork);
+                    $socialNetworkForm->submit($socialNetwork);
+                    $socialNetworkForm->getData()->setStartDate($socialNetwork['startDate']);
```

Comment:
Same here : store $socialNetworkForm->getData in a var and use it with fluent setters